src/server/debug/server/server.go

```diff
@@ -794,12 +794,11 @@ func (s *debugServer) queryLoki(ctx context.Context, queryStr string) ([]lokiLog
 	// of providing lines in chronological order even when the app uses both stdout and stderr.
 	var result []lokiLog
 
-	// FIXME: run once without an end, then work backwards
-	end := time.Now().Add(time.Minute)            // Account for some clock skew between our node and the Loki node.
+	var end time.Time
 	start := time.Now().Add(-30 * 24 * time.Hour) // 30 days.  (Loki maximum range is 30 days + 1 hour.)
 
-	for numLogs := 0; start.Before(end) && numLogs < maxLogs; {
-		resp, err := c.QueryRange(ctx, queryStr, serverMaxLogs, start, time.Time{}, "BACKWARD", 0, 0, true)
+	for numLogs := 0; end.IsZero() || start.Before(end) && numLogs < maxLogs; {
```

Comment:
should this be `(end.IsZero() || start.Before(end)) && numLogs < maxLogs`?

Reply:
Yes, yes it should be. Fixing.